packages/orchestration/src/exos/chain-hub.js

```diff
@@ -336,8 +362,53 @@ export const makeChainHub = (agoricNames, vowTools) => {
       // @ts-expect-error XXX generic parameter propagation
       return lookupChainsAndConnection(primaryName, counterName);
     },
+
+    /**
+     * Register an asset that may be held on a chain other than the issuing
+     * chain.
+     *
+     * @param {Denom} denom - on the holding chain, whose name is given in
+     *   `detail.chainName`
+     * @param {DenomDetail} detail - chainName and baseName must be registered
+     */
+    registerAsset(denom, detail) {
+      const { chainName, baseName } = detail;
+      chainInfos.has(chainName) ||
+        Fail`must register chain ${q(chainName)} first`;
+      chainInfos.has(baseName) ||
+        Fail`must register chain ${q(baseName)} first`;
+      denomDetails.init(denom, detail);
+    },
+    /**
+     * Retrieve holding, issuing chain names etc. for a denom.
+     *
+     * @param {Denom} denom
+     */
+    lookupAsset(denom) {
+      return denomDetails.get(denom);
+    },
   });
 
   return chainHub;
 };
 /** @typedef {ReturnType<typeof makeChainHub>} ChainHub */
+
+/**
+ * @param {ChainHub} chainHub
+ * @param {string} name
+ * @param {CosmosAssetInfo[]} assets
+ */
+export const registerAssets = (chainHub, name, assets) => {
+  for (const { base, traces } of assets) {
+    const native = !traces;
+    native || traces.length === 1 || Fail`unexpected ${traces.length} traces`;
+    const [chainName, baseName, baseDenom] = native
+      ? [name, name, base]
+      : [
+          name,
+          traces[0].counterparty.chain_name,
+          traces[0].counterparty.base_denom,
+        ];
```

Comment:
Is the plan to register a representation of an asset on every chain (that it's connected to)? And we'll only end up using `denomHash()` in tests / scripts to build proposals?

Reply:
Yes, that seems like the simplest thing that can work, based on discussion with @turadg earlier this week.

Maybe not every asset, but every asset they put in the chain registry.